db.go

```diff
@@ -460,6 +500,59 @@ func retentionCutoffDirs(dir string, mint int64) ([]string, error) {
 	return delDirs, nil
 }
 
+// maxBytesCutoffDirs returns all the oldest directories that would need to be
+// deleted to maintain the 'MaxBytes' option.
+func maxByteCutoffDirs(dir string, limit int64, metrics *dbMetrics) ([]string, error) {
+	df, err := fileutil.OpenDir(dir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "open directory")
+	}
+	defer df.Close()
+
+	dataDirSize, err := dirSize(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	delDirs := []string{}
+	if (dataDirSize <= limit) {
+		return delDirs, nil
+	}
+	// Limit has been exceeded
+	sizeToDelete := int64(0)
+
+	dirs, err := blockDirs(dir)
```

Comment:
I was under the impression that it did return blocks ordered by time range because in the code for retentionCutoffDirs it calls blockDirs, then passes through each block and adds it to the list of blocks to be deleted until it finds one that is after the cutoff time. There's a comment saying
```
// The first block we encounter marks that we crossed the boundary
// of deletable blocks.
if meta.MaxTime >= mint {
	break
}
```
If it isn't in order, wouldn't this also not function properly?

Reply:
Block directories are [ULID](https://github.com/oklog/ulid) which are lexicographically sortable identifiers.

Reply:
Ah great, didn't know that :)